lightning-invoice/src/payment.rs

```diff
@@ -217,7 +268,28 @@ pub trait Router<S: Score> {
 /// payments, if this is less than the total number of paths, we will never even retry all of the
 /// payment's paths.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub struct RetryAttempts(pub usize);
+pub enum Retry {
+	/// Count Out. Max number of attempts to retry payment.
+	///
+	/// Note that this is the number of *path* failures, not full payment retries. For multi-path
+	/// payments, if this is less than the total number of paths, we will never even retry all of the
+	/// payment's paths.
+	Attempts(usize),
+	#[cfg(feature = "std")]
+	/// Time Out. Max time spent to retry payment
```

Comment:
The phrasing is a little awkward here. How about "Time elapsed before abandoning retries for a payment."?

Reply:
yours is much better. Thanks!